Add a doc for how to use the msidentity-app-sync cli tool (#1027)
AzureAD/microsoft-identity-web: tools/app-provisioning-tool/vs2019-16.9-how-to-use.md

# Creating ASP.NET Core projects with Microsoft identity platform(Azure AD/B2C).

> In the 16.9 release of [Visual Studio](https://visualstudio.microsoft.com/downloads/), we updated the ASP.NET Core templates to allow you to use the Microsoft identity platform for authentication. We are still working to provide an experience where you can specify the auth options in Visual Studio 2019 and have it registered with the Microsoft identity platform and configure the generated project. With the new auth option, Microsoft identity platform, in the 16.9 release. You can create projects that are setup to for authentication, but there will be some additional work that you'll need to perform to complete the process. This document walks you through the available options . In a later update to Visual Studio we will streamline this experience so that all steps can be performed in Visual Studio itself.

Reviewer: ```suggestion
In the 16.9 release of [Visual Studio](https://visualstudio.microsoft.com/downloads/), we updated the ASP.NET Core templates to allow you to use the Microsoft identity platform for authentication. We are still working to provide an experience where you can specify the auth options in Visual Studio 2019 and have it registered with the Microsoft identity platform and configure the generated project. With the new auth option, Microsoft identity platform, in the 16.9 release, you can create projects that are setup for authentication, but there will be some additional work that you'll need to perform to complete the process. This document walks you through the available options . In a later update to Visual Studio we will streamline this experience so that all steps can be performed in Visual Studio itself.
```

The options are:

- Option 1: Go back to the previous experience that took care of all the steps (but with legacy limitations)

> - Option 2: Use a dotnet command line tool provided by Microsoft identity platform in 16.9 to register an app, and configure the code for you, or configure the code from an existing app. This tool also works if you created your project with the 1`dotnet new` command

Reviewer: ```suggestion
- Option 2: Use a dotnet command line tool provided by Microsoft identity platform in 16.9 to register an app, and configure the code for you, or configure the code from an existing app. This tool also works if you created your project with the `dotnet new` command
```

- [Azure PowerShell](https://docs.microsoft.com/en-us/powershell/azure/authenticate-azureps?view=azps-5.5.0)

Once you sign in with any of these tools, you can start using the identity sync tool. Let's look at how to use the tool to provision a new App Registration and configure the project. To get a list of the available built-in options run: `msidentity-app-sync –help` (or `msidentity-app-sync -h`). The current help output is shown in the image below.

![](images/tool-help.png)

From the help output, you can see that all the options are optional. You'll typically want to execute this command from the same directory that contains the project file (or solution in the case of Blazor WebAssembly hosted code) that you are trying to update. If you execute this command without any parameters, the following will occur (_assuming you are signed into a single Azure account_).

- The tool will detect what type of asp.net core app the project represents, the supported project types include; web (webapp and mvc templates), web API, Blazor Server and Blazor WebAssembly (hosted or not).
- The tool will detect which type of auth the project is configured for.

> - A new Azure AD App registration will be created in the [home tenant](https://docs.microsoft.com/en-us/azure/active-directory/develop/single-and-multi-tenant-apps#:~:text=Azure%20Active%20Directory%20(Azure%20AD,their%20security%20and%20operational%20policies.) of the signed in user

Reviewer 2: ```suggestion
- A new Azure AD App registration will be created in the [home tenant](https://docs.microsoft.com/azure/active-directory/develop/single-and-multi-tenant-apps#who-can-sign-in-to-your-app) of the signed in user
```
